### PR TITLE
replace basename() with path_extract_filename() in resolved-resolv-conf.c

### PR DESCRIPTION
**Pull request by stluo**
as titled

### CODE REVIEW

File: src/resolve/resolved-resolv-conf.c [hunk 1]
```diff
@@ -371,7 +373,11 @@ int manager_write_resolv_conf(Manager *m) {
 
                 temp_path_stub = mfree(temp_path_stub); /* free the string explicitly, so that we don't unlink anymore */
         } else {
-                r = symlink_atomic_label(basename(PRIVATE_UPLINK_RESOLV_CONF), PRIVATE_STUB_RESOLV_CONF);
+                r = path_extract_filename(PRIVATE_UPLINK_RESOLV_CONF, &fname);
+                if (r < 0)
+                        return r;
```
Comment (poettering) [inline src/resolve/resolved-resolv-conf.c]: please add a log message here, i.e.:

```c
if (r < 0) 
        return log_warning_errno(r, "Failed to extract filename from path '" PRIVATE_UPLINK_RESOLVE_CONF "', ignoring: %m");
```

**Comment by poettering** [comment]
looks good, just one addition please.

File: src/resolve/resolved-resolv-conf.c (hunk 1 above)
Comment (stluo) [reply src/resolve/resolved-resolv-conf.c]: fixed!

**Comment by bluca** [comment]
```
../src/resolve/resolved-resolv-conf.c: In function ‘manager_write_resolv_conf’:
../src/resolve/resolved-resolv-conf.c:378:94: error: expected ‘)’ before ‘PRIVATE_UPLINK_RESOLVE_CONF’
  378 |                         return log_warning_errno(r, "Failed to extract filename from path '" PRIVATE_UPLINK_RESOLVE_CONF "', ignoring: %m");
      |                                                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/basic/log.h:214:86: note: in definition of macro ‘log_full_errno_zerook’
  214 |                         ? log_internal(_level, _e, PROJECT_FILE, __LINE__, __func__, __VA_ARGS__) \
      |                                                                                      ^~~~~~~~~~~
../src/basic/log.h:253:41: note: in expansion of macro ‘log_full_errno’
  253 | #define log_warning_errno(error, ...)   log_full_errno(LOG_WARNING, error, __VA_ARGS__)
      |                                         ^~~~~~~~~~~~~~
../src/resolve/resolved-resolv-conf.c:378:32: note: in expansion of macro ‘log_warning_errno’
  378 |                         return log_warning_errno(r, "Failed to extract filename from path '" PRIVATE_UPLINK_RESOLVE_CONF "', ignoring: %m");
      |                                ^~~~~~~~~~~~~~~~~
../src/basic/log.h:214:39: note: to match this ‘(’
  214 |                         ? log_internal(_level, _e, PROJECT_FILE, __LINE__, __func__, __VA_ARGS__) \
      |                                       ^
../src/basic/log.h:229:17: note: in expansion of macro ‘log_full_errno_zerook’
  229 |                 log_full_errno_zerook(level, _error, __VA_ARGS__);      \
      |                 ^~~~~~~~~~~~~~~~~~~~~
../src/basic/log.h:253:41: note: in expansion of macro ‘log_full_errno’
  253 | #define log_warning_errno(error, ...)   log_full_errno(LOG_WARNING, error, __VA_ARGS__)
      |                                         ^~~~~~~~~~~~~~
../src/resolve/resolved-resolv-conf.c:378:32: note: in expansion of macro ‘log_warning_errno’
  378 |                         return log_warning_errno(r, "Failed to extract filename from path '" PRIVATE_UPLINK_RESOLVE_CONF "', ignoring: %m");
      |                                ^~~~~~~~~~~~~~~~~
```

File: src/resolve/resolved-resolv-conf.c [hunk 2]
```diff
@@ -371,7 +373,11 @@ int manager_write_resolv_conf(Manager *m) {
 
                 temp_path_stub = mfree(temp_path_stub); /* free the string explicitly, so that we don't unlink anymore */
         } else {
-                r = symlink_atomic_label(basename(PRIVATE_UPLINK_RESOLV_CONF), PRIVATE_STUB_RESOLV_CONF);
+                r = path_extract_filename(PRIVATE_UPLINK_RESOLV_CONF, &fname);
+                if (r < 0)
+                        return log_warning_errno(r, "Failed to extract filename from path '" PRIVATE_UPLINK_RESOLVE_CONF "', ignoring: %m");
```
Comment (poettering) [inline src/resolve/resolved-resolv-conf.c]: PRIVATE_UPLINK_RESOLVE_CONF → PRIVATE_UPLINK_RESOLV_CONF

(one "E" too much)

**Comment by stluo** [comment]
Is there a way to rerun some of the CI stuff? i don't think some of the errors are related but i could be wrong

File: src/resolve/resolved-resolv-conf.c [hunk 3]
```diff
@@ -328,6 +329,7 @@ int manager_write_resolv_conf(Manager *m) {
         _cleanup_ordered_set_free_ OrderedSet *dns = NULL, *domains = NULL;
         _cleanup_(unlink_and_freep) char *temp_path_uplink = NULL, *temp_path_stub = NULL;
         _cleanup_fclose_ FILE *f_uplink = NULL, *f_stub = NULL;
+        _cleanup_free_ char *fname = NULL;
```
Comment (poettering) [inline src/resolve/resolved-resolv-conf.c]: one more quest, can you move this to the innermost block this is used in? we generally try to minimize scope

**Comment by poettering** [comment]
thanks!

**Pull request merged**